Server/mods/deathmatch/logic/luadefs/CLuaDatabaseDefs.cpp

```diff
@@ -660,6 +661,29 @@ int CLuaDatabaseDefs::DbPrepareString(lua_State* luaVM)
     return 1;
 }
 
+int CLuaDatabaseDefs::DbGetConnectionQueueSize(lua_State* luaVM)
+{
+    CDatabaseConnectionElement* pElement;
+    CScriptArgReader argStream(luaVM);
+
+    argStream.ReadUserData(pElement);
+
+    if (!argStream.HasErrors())
+    {
+        int size = g_pGame->GetDatabaseManager()->GetQueueSizeFromConnection(pElement->GetConnectionHandle());
+        if (size >= 0)
+        {
+            lua_pushnumber(luaVM, size);
+            return 1;
+        }
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
```

Comment:
New API functions are now expected throw errors instead of print warnings.

```suggestion
        return luaL_error(luaVM, argStream.GetFullErrorMessage());
```